src/components/map/index.tsx

```diff
@@ -0,0 +1,59 @@
+import { useRef, useEffect } from 'react';
+import { Icon, Marker, layerGroup } from 'leaflet';
+import 'leaflet/dist/leaflet.css';
+
+import { Offer } from '../../types/offer';
+import useMap from '../../hooks/use-map';
+import { URL_MARKER_DEFAULT, URL_MARKER_CURRENT } from '../../const';
+
+type MapProps = {
+  city: Offer['city'];
+  points: Offer[];
+  selectedPoint: Offer | undefined;
+  className: string;
+};
+
+const defaultCustomIcon = new Icon({
+  iconUrl: URL_MARKER_DEFAULT,
+  iconSize: [40, 40],
+  iconAnchor: [20, 40]
+});
+
+const currentCustomIcon = new Icon({
+  iconUrl: URL_MARKER_CURRENT,
+  iconSize: [40, 40],
+  iconAnchor: [20, 40]
+});
+
+function Map({ city, points, selectedPoint, className }: MapProps): JSX.Element {
+  const mapRef = useRef(null);
+  const map = useMap(mapRef, city);
+
+  useEffect(() => {
+    if (map) {
+      const markerLayer = layerGroup().addTo(map);
+      points.forEach((point) => {
+        const marker = new Marker({
+          lat: point.location.latitude,
+          lng: point.location.longitude
+        });
+
+        marker
+          .setIcon(
+            selectedPoint !== undefined && point.id === selectedPoint.id
```

Comment:
Do we need strict comparison with `undefined`?. Maybe just check `selectedPoint && point.id === selectedPoint.id` or even `point.id === selectedPoint?.id` could work (optional chaining)